kimu/core/intersection_tool_lines.py

```diff
@@ -0,0 +1,72 @@
+from qgis.PyQt.QtCore import QVariant
+from qgis.core import (
+    QgsFeature,
+    QgsField,
+    QgsPointXY,
+    QgsProject,
+    QgsGeometry,
+    QgsVectorLayer,
+    QgsWkbTypes,
+)
+from qgis.utils import iface
+
+from ..qgis_plugin_tools.tools.custom_logging import setup_logger
+from ..qgis_plugin_tools.tools.i18n import tr
+from ..qgis_plugin_tools.tools.resources import plugin_name
+
+LOGGER = setup_logger(plugin_name())
+
+
+class IntersectionLines:
+
+    @staticmethod
+    def __check_valid_layer(layer: QgsVectorLayer) -> bool:
+        """Checks if layer is valid"""
+        if (
+            isinstance(layer, QgsVectorLayer)
+            and layer.isSpatial()
+            and layer.geometryType() == QgsWkbTypes.LineGeometry
+        ):
+            return True
+        return False
+
+    def run(self) -> None:
+        layer = iface.activeLayer()
+        if not self.__check_valid_layer(layer):
+            LOGGER.warning(tr("Please select a line layer"), extra={"details": ""})
+            return
+
+        if len(layer.selectedFeatures()) != 2:
+            LOGGER.warning(tr("Please select two line features"), extra={"details": ""})
+            return
+
+        vl = QgsVectorLayer("Point", "temp", "memory")
```

Comment:
Pay attention to giving more descriptive names for variables. `vl` may hint that this is a virtual layer, but something such as `intersection_layer` or `result_layer` would probably be better. I'll add some suggestions, but feel free to use other names.